include/stx/format_impl.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <string_view>
+#include <charconv>
+#include <optional>
+
+#include "formatter.h"
+
+namespace stx::format_impl
+{
+
+/**
+ * Helper for calling formatter<T>::format on the n-the tuple element.
+ */
+template <size_t>
+struct format_value_at_t;
+
+template <size_t _Index>
+struct format_value_at_t
+{
+    /* NOTE: _Index is 1based, Index0 is 0based */
+    static constexpr auto Index0 = _Index - 1u;
```

Comment:
Yes, _Index is the 1based-index, Index0 the same, but 0based, index.
Having _Index 1 based makes the template recursion _ways_ simpler (because of unsigned underflow for empty tuples).

Reply:
![jerry yes](https://i.pinimg.com/474x/45/f9/3c/45f93cc1c6c1e7d77cec6640feac86a2--rick-and-morty-jerry-oconnell.jpg)

Reply:
Checking at `(unsigned)-1` would be an option, but I prefer the 0u.